sycl/include/CL/sycl/detail/kernel_impl.hpp

```diff
@@ -29,118 +28,122 @@ class program_impl;
 
 class kernel_impl {
 public:
+  /// Constructs a SYCL kernel instance from a PiKernel
+  ///
+  /// This constructor is used for OpenCL interoperability. It always marks
+  /// kernel as being created from source and creates a new program_impl
+  /// instance.
+  ///
+  /// @param Kernel is a valid PiKernel instance
+  /// @param SyclContext is a valid SYCL context
   kernel_impl(RT::PiKernel Kernel, const context &SyclContext);
 
+  /// Constructs a SYCL kernel instance from a SYCL program and a PiKernel
+  ///
+  /// This constructor creates a new instance from PiKernel and associates it
+  /// with the provided SYCL program. If context of PiKernel differs from
```

Comment:
"and associates it with the provided SYCL program."
I'm not sure what does **associates** mean here. Reading the implementation, I see that we check that PiKernel belongs to `SyclContext`. Constructor just saves the pointer to the SYCL program and that's it. Is there missing logic to implement?
If not, I think this comment should be re-phrased.